.github/workflows/bench.yaml

```diff
@@ -0,0 +1,55 @@
+name: Benchmarks
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+
+jobs:
+  bench:
+    runs-on: ubuntu-22.04
+    steps:
```

Comment:
I think several of these could probably be improved upon by drawing from other helpful GitHub actions. For example, installing Rust and installing crate-binaries have pretty good (and more optimized) variants. The other workflow files may be able to provide some inspiration.

Reply:
- [x] migrated towards `dtolnay/rust-toolchain`
- [x] using lock files when building benchmarks
- [ ] add support of `taiki-e/install-action` to cargo-export